### DISCUSSION TITLE
[@types/express] 4 argument function is not inferred as ErrorRequestHandler in Express.use

### DISCUSSION THREAD

**Discussion by wirekang** [DefinitelyTyped/DefinitelyTyped#60353] [Issues with a @types package] [answered]
```typescript
import {ErrorRequestHandler} from "express";
import express from "express"

const errorHandler: ErrorRequestHandler = (err,req,res,next)=>{}

const app = express()

app.use(errorHandler) // Works well with typescript
app.use((err,req,res,next)=>{}) //     <-------  Wrong type here
```

All arguments(err~next) is `any` type. Is it intended?

[a comment by an automated account is not shown]

**Comment by alex-kinokon** [comment, marked as answer]
The problem is that [the signature using `RequestHandler` has a higher precedence over the one using `RequestHandlerParams`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ef87ee53bc501c0f0e79797add156fd8fa904ede/types/express-serve-static-core/index.d.ts#L178). If you swap that two lines it’s going the work. The problem was introduced by #10025.

Edit: [Pull request](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60792)
